### PR TITLE
Small Rcpp FAQ extension

### PR DESCRIPTION
**Pull request by eddelbuettel**
Addition of small section motivated by #1148

#### Checklist

- [X] Code compiles correctly
- [X] `R CMD check` still passes all tests (CI failure is spurious and caused by hub.docker.com)
- [ ] Prefereably, new tests were added which fail without the change
- [ ] Document the changes by file in [ChangeLog](https://github.com/RcppCore/Rcpp/blob/master/ChangeLog)

### CODE REVIEW

File: vignettes/rmd/Rcpp-FAQ.Rmd [hunk 1]
```diff
@@ -1765,3 +1765,13 @@ package, it is generally recommended to have both `Imports: Rcpp` and
 that this symbol is registered when your package is loaded by R, and as a
 side-effect certain other \pkg{Rcpp} function identifiers will also be
 resolved properly.
+
+## On macOS, 'no function function for call to `R_lsInternal`'
```
Comment (Enchufa2) [inline vignettes/rmd/Rcpp-FAQ.Rmd:1769]:
```suggestion
## On macOS, 'no matching function for call to `R_lsInternal`'
```

Comment (eddelbuettel) [reply vignettes/rmd/Rcpp-FAQ.Rmd:1769]: Whoops. Will fix -- great catch.

**Review by Enchufa2 — APPROVED** [review]

File: vignettes/rmd/Rcpp-FAQ.Rmd [hunk 2]
```diff
@@ -1765,3 +1765,12 @@ package, it is generally recommended to have both `Imports: Rcpp` and
...
 resolved properly.
+
+## On macOS, 'no matching function for call to `R_lsInternal`'
+
+In issue [#1148](https://github.com/RcppCore/Rcpp/issues/1148) an error due
+to overeager includes was reported. Including `Rinternals.h` along with the
+(macOS-only) `mach/boolean.h` lead to linker error as `mach/boolean`
+redefines `TRUE` leading to bad interactions with the `Rboolean` enum type. A
+very simple solution is to be more careful and conservative with `#include`
+files and a) have `#include <mach/boolean.h>` appear first and b) skip
```
Comment (kevinushey) [inline vignettes/rmd/Rcpp-FAQ.Rmd:1776]: Did this sentence end prematurely?

Comment (eddelbuettel) [reply vignettes/rmd/Rcpp-FAQ.Rmd:1776]: Uggh. I am having some minor fights with the md mode in emacs reverting. Will need to check.

Comment (eddelbuettel) [reply vignettes/rmd/Rcpp-FAQ.Rmd:1776]: Grrr. Yes. Lacks "$include <Rinternals.h>". My bad.

Comment (eddelbuettel) [reply vignettes/rmd/Rcpp-FAQ.Rmd:1776]: Suggest adding `the 

```
the `#include <Rinternals.h>` as it is included by `Rcpp.h` anyway.
```

**Pull request merged**